include/swift/AST/NameLookupRequests.h

```diff
@@ -324,6 +328,28 @@ class LookupPrecedenceGroupRequest
   bool isCached() const { return true; }
 };
 
+/// Expand the given ASTScope. Requestified to detect recursion.
+class ExpandASTScopeRequest
+    : public SimpleRequest<ExpandASTScopeRequest,
+                           ast_scope::ASTScopeImpl *(ast_scope::ASTScopeImpl *,
+                                                     ast_scope::ScopeCreator *),
+                           CacheKind::Uncached> {
```

Comment:
I think this is `SeparatelyCached`, because you can check `wasExpanded` in `ASTScopeImpl`.